File: snippets/python/40-users-groups/01-find-users.py
# import the shared client:  from elo_playground import connect
# topic:    List users (and groups)
# category: Users & groups
# id:       users.find-users

import os
from elo_playground import connect

# --- local ELO test box (override with ELOPG_* env vars or a .env) ---
ELO_BASE_URL = os.getenv("ELOPG_ELO_BASE_URL", "http://localhost:9090/ix-Repository1") # ELOPG_DEFAULT:base_url
ELO_USER = os.getenv("ELOPG_ELO_USER", "Administrator") # ELOPG_DEFAULT:user
ELO_PASS = os.getenv("ELOPG_ELO_PASSWORD", "elo") # ELOPG_DEFAULT:password

elo = connect(base_url=ELO_BASE_URL, user=ELO_USER, password=ELO_PASS)

users = elo.find_all(
    "findFirstUsers", "findNextUsers", "sortedResult",
    {"findUserInfo": {"onlyUsers": True}, "max": 200},
)

print(f"{len(users)} users")
for u in users:
    print(f"  [{u['id']:>5}] {u['name']:<16} {u.get('displayName', '')}")

File: snippets/node/40-users-groups/01-find-users.mjs
// import the shared client:  import { connect } from "elo-playground";
// topic:    List users (and groups)
// category: Users & groups
// id:       users.find-users

import { connect } from "elo-playground";

// --- local ELO test box (override with ELOPG_* env vars or a .env) ---
const ELO_BASE_URL = process.env.ELOPG_ELO_BASE_URL || "http://localhost:9090/ix-Repository1";
const ELO_USER = process.env.ELOPG_ELO_USER || "Administrator";
const ELO_PASS = process.env.ELOPG_ELO_PASSWORD || "";

const elo = await connect({ baseUrl: ELO_BASE_URL, user: ELO_USER, password: ELO_PASS });

const users = await elo.findAll(
  "findFirstUsers", "findNextUsers", "sortedResult",
  { findUserInfo: { onlyUsers: true }, max: 200 },
);

console.log(`${users.length} users`);
for (const u of users) {
  console.log(`  [${String(u.id).padStart(5)}] ${u.name.padEnd(16)} ${u.displayName || ""}`);
}
